inc/amp.php

```diff
@@ -0,0 +1,139 @@
+ * @param  array  $block Block attributes and information.
+ * @return string HTML block output.
+ */
+function convert_block_to_ampiframe( $output, $block ) {
+	if ( ! is_amp() ) {
+		return $output;
+	}
+
+	if ( 'core/html' === $block['blockName'] ) {
+		$pym_src_regex = '~data-pym-src=[\'"]([^\'"]*)~';
+		$is_match      = preg_match( $pym_src_regex, $block['innerHTML'], $matches );
+		if ( ! $is_match ) {
+			return $output;
+		}
+
+		$pym_src = $matches[1];
+		return get_pym_ampiframe( $pym_src );
+	} elseif ( 'pym-shortcode/pym' === $block['blockName'] ) {
+		$pym_src = $block['attrs']['src'];
+		return get_pym_ampiframe( $pym_src, $block['attrs'] );		
+	}
+
+	return $output;
+}
+add_action( 'render_block', __NAMESPACE__ . '\convert_block_to_ampiframe', 10, 2 );
+
+/**
+ * Handle pym.js shortcode on AMP pages.
+ *
+ * @since  1.3.2.3
+ * @param  string $output HTML shortcode output.
+ * @param  string $tag Shortcode tag.
+ * @param  array  $attributes Shortcode attributes.
+ * @return string HTML shortcode output.
+ */
+function convert_shortcode_to_ampiframe( $output, $tag, $attributes ) {
+	if ( ! is_amp() || 'pym' !== $tag ) {
+		return $output;
+	}
+
+	if ( empty( $attributes['src'] ) ) {
+		return $output;
+	}
+
+	return get_pym_ampiframe( $attributes['src'], $attributes );
+}
+add_action( 'do_shortcode_tag', __NAMESPACE__ . '\convert_shortcode_to_ampiframe', 10, 3 );
+
+/**
+ * Build an amp-iframe out of pym.js iframe source. This is a pretty solid solution until native pym.js AMP compatibility.
+ *
+ * @since  1.3.2.3
+ * @see    https://github.com/ampproject/amphtml/issues/22714
+ * @param  string $src iframe src.
+ * @return string AMP-iframe HTML.
+ */
+function get_pym_ampiframe( $src, $atts = array() ) {
+	$src_domain_parts  = parse_url( $src );
+	$site_domain_parts = parse_url( get_site_url() );
+
+	if ( ! $src_domain_parts ) {
+		return '';
+	}
+
+
+	/**
+	 * Filter pym_shortcode_default_class allows setting the default class on embeds.
+	 *
+	 * @param String $default
+	 * @return String the default class name
+	 */
+	$default_class   = apply_filters( 'pym_shortcode_default_class', 'pym' );
+	$shortcode_class = empty( $atts['class'] ) ? '' : esc_attr( $atts['class'] );
+	$gutenberg_class = empty( $atts['className'] ) ? '' : esc_attr( $atts['className'] );
+	$align           = empty( $atts['align'] ) ? '' : 'align' . esc_attr( $atts['align'] );
+	$actual_classes  = implode( ' ', array(
+		$default_class,
+		$shortcode_class,
+		$gutenberg_class,
+		$align,
+	) );
+	$id              = empty( $atts['id'] ) ? '' : esc_attr( $atts['id'] );
+
+	$sandbox = 'allow-scripts';
+	if ( strcasecmp( $src_domain_parts['host'], $site_domain_parts['host'] ) ) {
+		$sandbox .= ' allow-same-origin';
+	}
+
+	ob_start();
+	?>
+	<div class='<?php echo esc_attr( $actual_classes ); ?>' id='<?php echo esc_attr( $id ); ?>'>
+		<amp-iframe 
+			src='<?php echo esc_url( $src ); ?>'
+			layout='intrinsic'
+			width='1200'
+			height='1200'
```

Comment:
You may want to consider a `fixed-height` layout and to have the `height` set to 200. See examples at https://amp-iframe-pymjs-resizable.glitch.me/

Reply:
With this:

```diff
diff --git a/inc/amp.php b/inc/amp.php
index 30e289b..adc3656 100644
--- a/inc/amp.php
+++ b/inc/amp.php
@@ -105,9 +105,8 @@ function get_pym_ampiframe( $src, $atts = array() ) {
        <div class='<?php echo esc_attr( $actual_classes ); ?>' id='<?php echo esc_attr( $id ); ?>'>
                <amp-iframe 
                        src='<?php echo esc_url( $src ); ?>'
-                       layout='intrinsic'
-                       width='1200'
-                       height='1200'
+                       layout='fixed-height'
+                       height='200'
                        sandbox='<?php echo esc_attr( $sandbox ); ?>'
                        frameborder='0'
                        resizable
```

the presentation of embeds goes from this:

![Screenshot_2020-03-02 AMP Testing, modified nprapps example, x2 – Pym Shortcode Testing](https://user-images.githubusercontent.com/1754187/75711688-cea42300-5c94-11ea-90dc-fdd9fda97235.png)

to this:

![Screenshot_2020-03-02 AMP Testing, modified nprapps example, x2 – Pym Shortcode Testing(1)](https://user-images.githubusercontent.com/1754187/75711700-d4016d80-5c94-11ea-87e2-f24ba333e0a2.png)

👍 

I'll include this change when merging #62.